cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -45,12 +45,23 @@ namespace cudf {
  */
 class scalar {
  public:
-  virtual ~scalar()           = default;
-  scalar(scalar&& other)      = default;
-  scalar(scalar const& other) = default;
+  virtual ~scalar()      = default;
+  scalar(scalar&& other) = default;
+
   scalar& operator=(scalar const& other) = delete;
   scalar& operator=(scalar&& other) = delete;
 
+  /**
+   * @brief Construct a new scalar object by deep copying another.
+   *
+   * @param[in] other The scalar to copy.
+   * @param[in] stream CUDA stream used for device memory operations.
+   * @param[in] mr Device memory resource to use for device memory allocation
```

Comment:
Without having `[out]` parameter, we don't need to specify `[in]`.

Reply:
I did not add the [in].